transport/src/main/java/io/scalecube/transport/Address.java

```diff
@@ -0,0 +1,132 @@
+package io.scalecube.transport;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.base.Strings;
+import com.google.common.base.Throwables;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public final class Address {
```

Comment:
Address is clear and short name; ClusterMember is a mistake - you are right - I will rename it to just Member.